torchvision/utils.py

```diff
@@ -382,6 +389,116 @@ def draw_keypoints(
     return torch.from_numpy(np.array(img_to_draw)).permute(2, 0, 1).to(dtype=torch.uint8)
 
 
+# Flow visualization code adapted from https://github.com/tomrunia/OpticalFlow_Visualization
+@torch.no_grad()
+def flow_to_image(
+    flow: torch.Tensor,
+) -> torch.Tensor:
+
+    """
+    Converts the two-dimensional flow to an RGB Image.
+
+    Args:
+        flow (Tensor): Flow of shape (2, H, W) and dtype torch.float.
+
+    Returns:
+        img (Tensor(3, H, W)): Image Tensor where each color corresponds to a given flow direction.
+    """
+
+    if flow.ndim != 3 or flow.size(0) != 2:
+        raise ValueError(f"Input flow should have shape (2, H, W), got {flow.shape}.")
+
+    max_norm = torch.sum(flow ** 2, dim=0).sqrt().max()
+    epsilon = torch.finfo((flow).dtype).eps
+    normalized_flow = flow / (max_norm + epsilon)
+    return _normalized_flow_to_colors(normalized_flow)
+
+
+@torch.no_grad()
+def _normalized_flow_to_colors(normalized_flow: torch.Tensor) -> torch.Tensor:
+
+    """
+    Applies the flow color wheel to (possibly clipped) flow components u and v.
+
+    Args:
+        normalized_flow (torch.Tensor): Normalized flow tensor of shape (2, H, W)
+    Returns:
+       img (Tensor(3, H, W)): Flow visualization image of dtype uint8.
+    """
+
+    u = normalized_flow[0, :, :]
+    v = normalized_flow[1, :, :]
+    flow_image = torch.zeros((3, u.shape[0], u.shape[1]), dtype=torch.uint8)
+    colorwheel = _make_colorwheel()  # shape [55x3]
+    num_cols = colorwheel.shape[0]
+    rad = torch.sqrt(torch.square(u) + torch.square(v))
+    a = torch.atan2(-v, -u) / torch.pi
+    fk = (a + 1) / 2 * (num_cols - 1)
+    k0 = torch.floor(fk).to(torch.long)
+    k1 = k0 + 1
+    k1[k1 == num_cols] = 0
+    f = fk - k0
+
+    for i in range(colorwheel.shape[1]):
+        tmp = colorwheel[:, i]
+        col0 = tmp[k0] / 255.0
+        col1 = tmp[k1] / 255.0
+        col = (1 - f) * col0 + f * col1
+        idx = rad <= 1
```

Comment:
Because this function takes the normalized flow as input, I believe that `rad` (actually the 2-norm) is  guaranteed to always be `<= 1`. So I feel like we might be able to drop this `idx` logic altogether?

Reply:
Simplified this entire logic.